server/lib/user/post/controller.ts

```diff
@@ -3,14 +3,22 @@ import {postItem} from "../../shared/database";
 import {saveUser} from "./service";
 import {handleResult} from "../../shared/result-handler";
 import {handleError} from "../../shared/error-handling";
+import {userValidationSchema} from "../model/register-user";
 
 export async function handler(event: APIGatewayEvent, _: Context, callback: APIGatewayProxyCallback) {
     try {
-        // Get user data from request body
+
         const requestBody = JSON.parse(event.body || '{}');
-        const email = requestBody.email;
 
-        saveUser(email);
+        // Validate request body
+        const { error } = userValidationSchema.validate(requestBody);
+        if (error) {
+            return handleError(callback, new Error(`Validation error: ${error.details[0].message}`));
```

Comment:
Instead of returning error jsut throw new error, let's keep handleError and HandleResult method inside controller files.

Reply:
throw new BadRequestException(`Validation error: ${error.details[0].message}`), later we would need to make sure the validaiton could reused and error would be displayed for all valdiated fields.